include/meta/meta.hpp

```diff
@@ -925,23 +939,88 @@ namespace meta
     /// \endcond
 
     /// Return a new \c meta::list constructed by transforming all the elements
-    /// in \p List with the unary Metafuncion Class Fun. \c transform can also
-    /// be called with two lists of the same length and a binary Metafunction
-    /// Class, in which case it returns a new list constructed with the results
-    /// of calling \c Fun with each element in the lists, pairwise.
+    /// in \p List with the unary Metafuncion Class \p Fun. \c transform can
+    /// also be called with two lists of the same length and a binary
+    /// Metafunction Class, in which case it returns a new list constructed with
+    /// the results of calling \c Fun with each element in the lists, pairwise.
     /// \par Complexity
     /// \f$ O(N) \f$.
     template <typename... Args>
     using transform = eval<detail::transform_<Args...>>;
 
+    ////////////////////////////////////////////////////////////////////////////
+    // filter
+    /// \cond
+    namespace detail
+    {
+        template <typename Predicate> struct filter_
+        {
+            template <typename State, typename A>
+            using apply = meta::if_<meta::apply<Predicate, A>,
+                                    meta::push_back<State, A>, State>;
+        };
+    } // namespace detail
+    /// \endcond
+
+    /// Returns a new meta::list where only those elements of \p List A that
+    /// satisfy the Metafunction Class \p Predicate such that
+    /// `apply<Pred,A>::%value` is \c true are present. That is, those elements
+    /// that don't satisfy the \p Predicate are "removed".
+    template <typename List, typename Predicate>
+    using filter = meta::foldl<List, meta::list<>, detail::filter_<Predicate>>;
+
+    ////////////////////////////////////////////////////////////////////////////
+    // static_const
+    ///\cond
+    namespace detail
+    {
+        template <typename T> struct static_const
+        {
+            static constexpr T value{};
+        };
+
+        // Avoid potential ODR violations with global objects:
+        template <typename T> constexpr T static_const<T>::value;
+    } // namespace detail
+
+    ///\endcond
+
+    ////////////////////////////////////////////////////////////////////////////
+    // for_each
+    /// \cond
+    namespace detail
+    {
+        struct for_each_fn
+        {
+            template <class UnaryFunction, class... Args>
+            constexpr auto operator()(meta::list<Args...>,
+                                      UnaryFunction &&f) const
+              -> decltype(std::forward<UnaryFunction>(f))
+            {
+                return (void)std::initializer_list<int>{
+                         (std::ref(f)(Args{}), 0)...},
```

Comment:
I was wrong about `std::ref`in my tweat -- it isn't needed here.. We should drop it and remove the `#include <functional>`